style/redux/README.md

```diff
@@ -0,0 +1,195 @@
+`createAction` is less readable and doesn't let us easily export named
+functions. Note that only `type` is mandatory.
+
+So far the common practice is to group files by type rather than by feature, and
+this style guide follows that pattern. But there's an alternative pattern -
+[ducks] - a proposal to bundle action creators with their reducers into a single
+file. Such a bundle is called a 'duck'. Its goal is to move towards a modular
+application structure.
+
+[redux-promise-middleware]: https://github.com/pburtchaell/redux-promise-middleware
+[redux-thunk]: https://github.com/gaearon/redux-thunk
+[Flux Standard Action]: https://github.com/acdlite/flux-standard-action
+[redux-actions]: https://github.com/acdlite/redux-actions
+[ducks]: https://github.com/erikras/ducks-modular-redux
+
+## Action types
+
+- Put action types in one file of constants.
+- Group related action types in an object.
+- Expose each object as a named export.
+- Expose all action types as the default export.
+
+```js
+// constants/actionTypes.js
+export const todo = {
+  fetch: 'FETCH_TODO',
+  set: 'SET_TODO',
+};
+
+export default {
+  todo,
+};
+```
+
+Action types identify the nature of the action that has occurred. Two actions
+with the same type MUST be strictly equivalent (using ===). By convention, type
+is usually a string constant or a Symbol.
+
+## Reducers
+
+- Expose reducers as the default export.
+- Expose handlers as named exports for unit testing.
+- Use constants instead of inline strings for action types.
+- Always define an `INITIAL_STATE` variable.
+
+```js
+// using redux-promise-middleware
+// using type-to-reducer
+
+// reducers/todo.js
+import typeToReducer from 'type-to-reducer';
+import { todo } from '../constants/actionTypes';
+
+const INITIAL_STATE = {
+  data: {},
+  error: false,
+  isRequesting: false,
+};
+
+export function addTodo(state, action) {
+  return { ...state, data: action.payload };
+}
+
+export const addTodoAsync = {
+  PENDING: (state) => ({
+    ...state,
+    isRequesting: true,
+  }),
+  REJECTED: (state, action) => ({
+    ...state,
+    isRequesting: false,
+    error: action.payload,
+  }),
+  FULFILLED: (state, action) => ({
+    ...state,
+    isRequesting: false,
+    data: action.payload,
+  }),
+};
+
+export default typeToReducer(
+  {
+    [todo.add]: addTodo,
+    [todo.addAsync]: addTodoAsync,
+  },
+  INITIAL_STATE
+);
+```
+
+The final reducer should be exposed as the default export. Individual handler
+functions should be exposed as named exports in order to simplify unit testing.
+
+The simplest way to create a reducer is to use the `handleActions` method of
+[redux-actions]. This avoids having to write a switch statement and a default
+handler. Another benefit is that it enforces the use of FSA-compliant action
+objects. Alternatively, use [type-to-reducer] for simplifying handling of async
+actions, as it allows nesting of related actions.
```

Comment:
Your example doesn't suggest how it solves one of the problems I highlighted and that this approach tries to solve - derived actions, ie, an async action `SET_EVENT` which spawns 3 new actions (using suffixes): `SET_EVENT_PENDING`, `SET_EVENT_FULFILLED`, `SET_EVENT_REJECTED`.
Using your approach you'd have to type a lot more things as I wrote in the previous comment.

Also, you're dealing with only 1 async action in the examples. I bet that in the apps you've built there was a bunch of actions with similar handlers, and those actions are probably async, so if you don't have an easier way to share code, you'll end up writing too much repeated code.
I have this example in oncostats to exemplify what I mean:
```js
import typeToReducer from 'type-to-reducer';
import { currentUser } from '../constants/actionTypes';

const INITIAL_STATE = {
  isRequesting: false,
  error: false,
  data: {},
};

const reducersByPrefix = {
  PENDING: (state) => ({
    ...state,
    isRequesting: true,
  }),
  REJECTED: (state, action) => ({
    ...state,
    isRequesting: false,
    error: action.payload,
  }),
  FULFILLED: (state, action) => ({
    ...state,
    isRequesting: false,
    data: action.payload,
  }),
};

export default typeToReducer(
  {
    [currentUser.fetch]: reducersByPrefix,
    [currentUser.set]: reducersByPrefix,
  },
  INITIAL_STATE
);
```

One can argue that typing is not cumbersome enough that it doesn't justify adding this tool. But don't forget that for almost every new bundle added to our redux app you have to manually write actions, action types and reducers, so any help towards minimizing that pain should be considered.

What are the actual problems you identify. And what would you suggest for getting the best of both worlds.
I've already experienced vanilla redux and it really bored me to create new things and manage others that were already written.